### PR TITLE
test(tab-button): add focus state screenshots

### PR DESCRIPTION
**Pull request by liamdebeasi**
Issue number: N/A

---------

## What is the current behavior?

We do not have screenshots that verify the focus states are being applied correctly.

## What is the new behavior?

- Added screenshot tests to verify focus states are being captured for non-translucent tab bars (iOS and MD) as well as translucent tab bars (iOS only)

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by brandyscarney — CHANGES_REQUESTED** [review]
I'm not really seeing a difference between the translucent and non-translucent tabs - should we add some divs behind them or something?

File: core/src/components/tab-button/test/states/tab-button.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            <ion-tab-button href="#" class="ion-focused">
+              <ion-label>Favorites</ion-label>
+            </ion-tab-button>
+          </ion-tab-bar>
+        `,
+          config
+        );
+
+        const tabBar = page.locator('ion-tab-bar');
+        await expect(tabBar).toHaveScreenshot(screenshot('tab-bar-focus-default-theme-no-translucency'));
```
[the remark this answers is not in the document]
Comment (liamdebeasi) [reply core/src/components/tab-button/test/states/tab-button.e2e.ts]: Fixed in https://github.com/ionic-team/ionic-framework/pull/27731/commits/72186539be532c14450271e2004fa833e4c57e42

File: core/src/components/tab-button/test/states/tab-button.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            <ion-tab-button href="#" class="ion-focused">
+              <ion-label>Favorites</ion-label>
+            </ion-tab-button>
+          </ion-tab-bar>
+        `,
+          config
+        );
+
+        const tabBar = page.locator('ion-tab-bar');
+        await expect(tabBar).toHaveScreenshot(screenshot('tab-bar-focus-custom-theme-no-translucency'));
```
[the remark this answers is not in the document]
Comment (liamdebeasi) [reply core/src/components/tab-button/test/states/tab-button.e2e.ts]: Fixed in https://github.com/ionic-team/ionic-framework/pull/27731/commits/bd1105cb44cb52c4da38d3e68fe98af391774de4

**Comment by liamdebeasi** [comment]
I added a dark background color in https://github.com/ionic-team/ionic-framework/pull/27731/commits/af08359fb0f3367875241bbdd1d7eee85bef1495. Screenshots are updating now.

**Review by brandyscarney — APPROVED** [review]

**Pull request merged**